easybuild/easyblocks/c/comsol.py

```diff
@@ -0,0 +1,78 @@
+##
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+##
+"""
+EasyBuild support for installing COMSOL, implemented as an easyblock
+
+@author: Mikael OEhman (Chalmers University of Technology)
+"""
+
+import re
+import shutil
+import os
+import stat
+
+from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import read_file, write_file
+from easybuild.tools.run import run_cmd
+
+
+class EB_COMSOL(PackedBinary):
+    """Support for installing COMSOL."""
+
+    def __init__(self, *args, **kwargs):
+        super(EB_COMSOL, self).__init__(*args, **kwargs)
+        self.configfile = os.path.join(self.builddir, 'my_setupconfig.ini')
+
+    def configure_step(self):
+        license = os.getenv('EB_COMSOL_LICENSE')
+        if license is None:
+            raise EasyBuildError("EB_COMSOL_LICENSE not set (required)")
```

Comment:
I like the approach in #1317 better, since that leverages the available `find_flexlm_license` function

Reply:
@boegel but it would also limit the license to just be flex licenses, which is not the only option for COMSOL, e.g. passcode, but I suppose that's basically unheard of in HPC environments.